File: python/basic_chat.py
"""
NeuronGate — Basic Chat Completion
"""
from openai import OpenAI

client = OpenAI(
    base_url="https://neurongate.net/v1",
    api_key="ng-your-api-key"
)

response = client.chat.completions.create(
    model="openai/gpt-4o",
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "What are the benefits of using an API gateway for AI?"}
    ],
    temperature=0.7,
    max_tokens=500
)

print(response.choices[0].message.content)
print(f"\nTokens: {response.usage.total_tokens}")

File: javascript/basic_chat.js
/**
 * NeuronGate — Basic Chat Completion (Node.js)
 * 
 * npm install openai
 */
import OpenAI from 'openai';

const client = new OpenAI({
  baseURL: 'https://neurongate.net/v1',
  apiKey: 'ng-your-api-key',
});

async function main() {
  const response = await client.chat.completions.create({
    model: 'anthropic/claude-sonnet-4-6',
    messages: [
      { role: 'system', content: 'You are a helpful assistant.' },
      { role: 'user', content: 'Explain microservices in 3 sentences.' },
    ],
    max_tokens: 200,
  });

  console.log(response.choices[0].message.content);
  console.log(`\nTokens: ${response.usage.total_tokens}`);
}

main();
